### ISSUE TITLE
Add ability to change profiles (~/.ssh/config files)

### ISSUE THREAD

**Issue by mattwilmott** [moul/advanced-ssh-config#258]
I work in a large cloud infrastructure provider and I often work in environments where ip ranges overlap. DNS resolution is not always an option either.

It would be nice to be able to setup multiple assh.yml files (and corresponding assh.d dirs) and have a means of switching between them(environments). ie 'assh load dev-lab4-tenantE' to load an environment definition, 'assh load default' to load the default assh.yml config etc

Currently this can be done via 'ssh -F' but making it persistent till the next change would be ideal, a bit like a symlink or similar. This would save having to add a flag to the command every invocation. I've also tried aliasing the command but our environment runs into the hundreds so it becomes unwieldly managing aliases at that rate

Also adding the ability to visually identify the current context would be helpful.

I don't suppose this is a common requirement today but as more people continue down the path of overlay networks etc in the future I envisage it will be!

**Comment by moul** [comment]
Hi @mattwilmott, thank you for your feedback

I like the idea of adding an easy way to switch from a config file to another with persistence

Right now, my preference would go for another way of achieving the same result without needing to switch between different config files but instead, declaring a way of switching between different namespaces of a same configuration; this way, we should be able to automatically switch between namespaces based on an automatic rule 

I will think a little bit more about this, and will fallback to the first idea if I can't find a good implementation for the second option

In the meantime, you can try to create your own script that will do something like this:

```shell
#!/bin/sh

ln -f ~/.ssh/$1.assh.yml ~/.ssh/assh.yml
assh config build > ~/.ssh/config
```

And then create as much `~/.ssh/<environment>.assh.yml`

**Comment by mattwilmott** [comment]
Yeah atm I am using a small bash script along with GNU Stow to move the symlinks about. It works but isnt ideal

Namespaces is essentially what we're aiming for ;) We just need to have a means of switching profile/namespace per invocation as well as being able to set the default profile/namespace in the event none is specified.

A bit like:

assh lab1 10.0.0.1 .  <- invokes namespace lab1 that maps to ~/.ssh/assh.d/lab1/* and connects to host 10.0.0.1

assh namespace lab1 <- sets namespace for subsequent calls to the lab1 namespace
assh namespace  <- prints the current namespace
> lab1
assh 10.0.0.1  <- that relies on the current set namespace, lab1

Finally maybe shorten or provide an alias for 'namespace' such as 'ns'
ie 'assh ns lab1' to set lab1 namespace

Cheers 
Matt
